src/components/FormPills/FormPills.vue

```diff
@@ -77,13 +97,19 @@ const splitItem = (item: string) => item.split(props.separator);
 const removeItem = (index: number) => items.value.splice(index, 1);
 
 const addItem = (item: string) => {
-    const newItems = splitItem(item).filter(v => v !== '');
+    if (items.value.length >= props.maxItems) {
+        return false;
+    }
+
+    const newItems = splitItem(item).filter(v => v.match(props.valueMatcher));
```

Comment:
Looks like work not that well (or my regex is wrong). Used `/^a$/` as regex but will not add new item.
![image](https://user-images.githubusercontent.com/45201651/149764384-999d5386-b1bf-44bb-b711-ebe8ccab80e0.png)

Reply:
[:) ❤️ ](https://code.tutsplus.com/tutorials/you-dont-know-anything-about-regular-expressions-a-complete-guide--net-7869)